### ISSUE TITLE
[comma-dangle] does not check enum declarations

### ISSUE THREAD

**Issue by lucaswerkmeister** [typescript-eslint/typescript-eslint#710]
**Repro**

```JSON
{
  "rules": {
    "comma-dangle": ["error", "always-multiline"]
  }
}
```

```TS
enum MyEnum {
    MY_VALUE,
    MY_OTHER_VALUE
}
```

**Expected Result**

Some kind of error on line three of the test file, since there is no trailing comma after the final enum member.

**Actual Result**

No error is reported.

**Additional Info**

Ideally, this would happen with no extra config, where typescript-eslint would somehow “plug into” eslint’s [comma-dangle](https://eslint.org/docs/rules/comma-dangle) rule. That said, its [current implementation](https://github.com/eslint/eslint/blob/93633c2b3716b17816bcb3dc221c49b75db41317/lib/rules/comma-dangle.js) doesn’t look like it invites this; for example, it [immediately throws away option members other than `arrays`/`objects`/`imports`/`exports`/`functions`](https://github.com/eslint/eslint/blob/93633c2b3716b17816bcb3dc221c49b75db41317/lib/rules/comma-dangle.js#L42-L71), so it’s not clear how typescript-eslint could have separate options for `enums` here. A separate, TypeScript-specific rule seems more likely (I’m not sure if there are any other syntax constructs besides enums that support dangling commas). In that case, of course, the separate rule would need to be mentioned in the repro to make it work – but I couldn’t find any existing rule that would provide this.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.0.1` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.9.0` |

**Comment by lolmaus** [comment]
For me, `"comma-dangle": ["error", "always-multiline"]` won't even enforce trailing comma in multiline arrays. 😒
